rust/connlib/tunnel/src/client.rs

```diff
@@ -444,28 +448,28 @@ impl ClientState {
         // We read this here to prevent problems with the borrow checker
         let is_dns_resource = self.is_dns_resource(&resource);
 
-        let Some(peer) = peer_by_resource_mut(&self.resources_gateways, &mut self.peers, resource)
-        else {
-            self.on_not_connected_resource(resource, &dst, now);
-            return None;
-        };
-
         let packet = maybe_mangle_dns_query_to_cidr_resource(
             packet,
             &self.dns_mapping,
             &mut self.mangled_dns_queries,
             now,
         );
 
+        let Some(peer) = peer_by_resource_mut(&self.resources_gateways, &mut self.peers, resource)
+        else {
+            self.on_not_connected_resource(resource, &dst, packet.as_immutable(), now);
+            return None;
+        };
+
+        let gid = peer.id();
+
         // Allowed IPs will track the IPs that we have sent to the gateway along with a list of ResourceIds
         // for DNS resource we will send the IP one at a time.
         if is_dns_resource && peer.allowed_ips.exact_match(dst).is_none() {
-            let gateway_id = peer.id();
-            self.send_proxy_ips(&dst, resource, gateway_id);
-            return None;
```

Comment:
If we have a tunnel, `boringtun` will buffer them internally for us.

Reply:
but the packets will reach the gateway even if they don't have the proxy ip, and the gateway will just discard them.

Reply:
Ah yes. I needed this because the tests were failing otherwise because there, delivery of the portal events is instant and the `Transmit`s have a latency to them.

Reply:
ah gotcha, this will just emit some packets on the gateway for the proxy ip